.drone.yml

```diff
@@ -1,109 +1,302 @@
-  #   commands:
-  #     - go build -v -ldflags "-X main.build=${DRONE_BUILD_NUMBER}" -a -o release/windows/amd64/drone-crowdin
-
-  publish_linux_amd64:
-    image: plugins/docker:17.05
-    pull: true
-    repo: jonasfranz/crowdin
-    tags: [ latest, 1.0.0, 1.0, 1 ]
-    secrets: [ docker_username, docker_password ]
-    dockerfile: Dockerfile
-    when:
-      branch: master
-      event: push
-
-  publish_linux_arm64:
-    image: plugins/docker:17.05
-    pull: true
-    repo: jonasfranz/crowdin
-    tags: [ linux-arm64 ]
-    secrets: [ docker_username, docker_password ]
-    dockerfile: Dockerfile.arm64
-    when:
-      branch: master
-      event: push
-
-  publish_linux_arm:
-    image: plugins/docker:17.05
-    pull: true
-    repo: jonasfranz/crowdin
-    tags: [ linux-arm ]
-    secrets: [ docker_username, docker_password ]
-    dockerfile: Dockerfile.arm
-    when:
-      branch: master
-      event: push
-
-  # publish_windows_amd64:
-  #   image: plugins/docker:17.05
-  #   pull: true
-  #   repo: jonasfranz/crowdin
-  #   tags: [ windows-amd64 ]
-  #   secrets: [ docker_username, docker_password ]
-  #   dockerfile: Dockerfile.windows
-  #   when:
-  #     branch: master
-  #     event: push
-
-  microbadger:
-    image: plugins/webhook:1
-    pull: true
-    secrets: [ webhook_url ]
-    when:
-      branch: master
-      event: push
-      status: [ success ]
+---
+kind: pipeline
+name: testing
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+- name: vet
+  pull: always
+  image: golang:1.13
+  commands:
+  - go vet -mod=vendor ./...
+  environment:
+    GO111MODULE: on
+
+- name: test
+  pull: always
+  image: golang:1.13
+  commands:
+  - go test -mod=vendor -cover $(go list -mod=vendor ./... | grep -v /vendor/)
+  environment:
+    GO111MODULE: on
+
+trigger:
+  branch:
+  - master
+
+---
+kind: pipeline
+name: linux-amd64
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+- name: build-push
+  pull: always
+  image: golang:1.13
+  commands:
+  - "go build -mod=vendor -v -ldflags \"-X main.build=${DRONE_BUILD_NUMBER}\" -a -o release/linux/amd64/drone-crowdin"
+  environment:
+    CGO_ENABLED: 0
+    GO111MODULE: on
+  when:
+    event:
+    - push
+    - pull_request
+
+- name: build-tag
+  pull: always
+  image: golang:1.13
+  commands:
+  - "go build -mod=vendor -v -ldflags \"-X main.version=${DRONE_TAG##v} -X main.build=${DRONE_BUILD_NUMBER}\" -a -o release/linux/amd64/drone-crowdin"
+  environment:
+    CGO_ENABLED: 0
+    GO111MODULE: on
+  when:
+    event:
+    - tag
+
+- name: executable
+  pull: always
+  image: golang:1.13
+  commands:
+  - ./release/linux/amd64/drone-crowdin --help
+
+- name: dryrun
+  pull: always
+  image: plugins/docker:linux-amd64
+  settings:
+    dockerfile: docker/Dockerfile.linux.amd64
+    dry_run: true
+    password:
+      from_secret: docker_password
+    repo: jonasfranz/drone-crowdin
+    tags: linux-amd64
+    username:
+      from_secret: docker_username
```

Comment:
Do we need the username, since `docker_username` is not available for pull requests?